hrrecruitment/CRM/HRRecruitment/DAO/HRVacancy.php

```diff
@@ -165,67 +125,64 @@ class CRM_HRRecruitment_DAO_HRVacancy extends CRM_Core_DAO
    * @var int unsigned
    */
   public $created_id;
-
   /**
    * class constructor
    *
-   * @access public
    * @return civicrm_hrvacancy
    */
-  function __construct()
-  {
+  function __construct() {
     $this->__table = 'civicrm_hrvacancy';
     parent::__construct();
   }
   /**
-   * return foreign keys and entity references
+   * Returns foreign keys and entity references
    *
-   * @static
-   * @access public
-   * @return array of CRM_Core_Reference_Interface
+   * @return array
+   *   [CRM_Core_Reference_Interface]
    */
-  static function getReferenceColumns()
-  {
-    if (!self::$_links) {
-      self::$_links = array(
-        new CRM_Core_Reference_Basic(self::getTableName() , 'created_id', 'civicrm_contact', 'id') ,
-      );
+  static function getReferenceColumns() {
+    if (!isset(Civi::$statics[__CLASS__]['links'])) {
+      Civi::$statics[__CLASS__]['links'] = static ::createReferenceColumns(__CLASS__);
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName() , 'created_id', 'civicrm_contact', 'id');
+      CRM_Core_DAO_AllCoreTables::invoke(__CLASS__, 'links_callback', Civi::$statics[__CLASS__]['links']);
     }
-    return self::$_links;
+    return Civi::$statics[__CLASS__]['links'];
```

Comment:
You expect Civi to improve it's code base and then they come up with things like this (╯°□°）╯︵ ┻━┻